### DISCUSSION TITLE
nginx-helper plugin conflicts with feed import progress bar?

### DISCUSSION THREAD

**Discussion by brotskydotcom2** [code4recovery/12-step-meeting-list#1267] [Q&A] [answered]
We are trying to move our site from DreamHost's standard shared hosting to their managed shared hosting (DreamPress) plan.  Their standard shared hosting uses Apache all the way, but their managed plans put nginx in front of Apache.  They do their caching on the nginx side by using [this version of the nginx-helper plugin](https://github.com/rtCamp/nginx-helper).  The problem we're having is that there is some sort of interference between the admin-ajax.php GET that updates the progress bar and the nginx-helper, with the result being that we don't see any progress reported.  Disabling the nginx-helper plugin when we refresh feeds and then re-enabling it after is an effective workaround.

Has anyone encountered something like this before?  Does anyone have any clues about what the source of the conflict is?

**Comment by joshreisner** [comment]
hm, sorry to hear there's an incompatibility there. are other calls to `admin-ajax.php` failing? for example looking up addresses / locations / groups on the meeting edit page?

there's nothing fundamentally incompatible between nginx and the importer, the site we use to test the importer runs on nginx (with no apache).

just so i understand the issue fully, what's the benefit of keeping the plugin enabled? could it be selectively disabled for `admin-ajax.php`?

**Reply by brotskydotcom2** [reply]
So the `nginx-helper` plugin is a cache-clearer that watches page updates and clears the appropriate parts of the cache.  (It also handles permalinks and suchlike, probably fonts too, not completely sure.). The support team at DreamHost says they aren't sure what the compatibility issue is, so they've switched our site to using Varnish and the "Proxy Cache Purge" plugin.  That seems to have resolved the issue.

To answer your question: no, the other calls to admin-ajax.php have no issues.  My _suspicion_ is that the nginx-helper decides how long it's going to wait for an uncached page to respond, and when that gets past a certain threshold it serves the page from the cache.  Since our feed runs off a Heroku eco dyno, it often takes 5 seconds to spin up and that exceeds the threshold that nginx-helper will allow for the response.  Even the normal import time seems to have cut it close.

Please feel free to close this question, @joshreisner, and thanks for your help!

**Comment by brotskydotcom2** [comment, marked as answer]
So I believe I've gotten to the bottom of this issue with `nginx-helper`.  The import process works by deleting all the existing posts from the feed, then recreating them.  The `nginx-helper` plugin is monitoring these deletions and synchronously trying to clear the nginx fast-cgi cache of all these posts.  That takes quite a while and eventually causes things to timeout.

There is another plugin, `LifterLMS`, that [reported this issue](https://github.com/rtCamp/nginx-helper/issues/235) to the maintainers of `nginx-helper` back in 2020, with no response. (See their internal issue [here](https://github.com/gocodebox/lifterlms/issues/1073).)  They eventually fell back to the workaround that I am using, which is to disable the plugin when refreshing the feed and then manually purging the cache.  Kind of a drag.
